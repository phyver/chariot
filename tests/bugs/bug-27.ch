:set incremental_SCP false
:set show_all_steps true
:set show_bad_loops true
:set depth 3

data nat where Zero : nat | Succ : nat -> nat

data list('x) where Nil : list('x) | Cons : 'x -> list('x) -> list('x)

val length [] = 0
  | length x::xs = (length xs) + 1
