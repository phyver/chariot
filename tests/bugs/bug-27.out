>> option verbose is set to 1
>> option incremental_SCP is set to false
>> option show_all_steps is set to true
>> option show_bad_loops is set to true
>> option depth is set to 3
>> inductive type nat was succesfully defined
>> inductive type list was succesfully defined
>> Typing for length successful
>> the definition for length is complete
>> test at depth 3
>> Graph of paths at iteration 0
>> calls from length to length:
>>     length, x₁::x₂ => length, x₂, .<<1¹>>

>> Graph of paths at iteration 1
>> calls from length to length:
>>     length, x₁::x₂ => length, x₂, .<<1¹>>
>>     length, x₁::x₂::x₃ => length, x₃, .<<∞¹>>

>> Graph of paths at iteration 2
>> calls from length to length:
>>     length, x₁::x₂ => length, x₂, .<<1¹>>
>>     length, x₁::x₂::x₃ => length, x₃, .<<∞¹>>
>>     length, x₁::x₂::x₃::x₄ => length, x₄, .<<∞¹>>

>> Graph of paths at iteration 3
>> calls from length to length:
>>     length, x₁::x₂ => length, x₂, .<<1¹>>
>>     length, x₁::x₂::x₃ => length, x₃, .<<∞¹>>
>>     length, x₁::x₂::x₃::x₄ => length, x₄, .<<∞¹>>
>>     length, x₁::x₂::x₃::x₄ => length, <-1¹>x₄, .<<∞¹>>

>> the definition length is provably correct

